chronosQ-reservation-impl-redis/src/main/java/flipkart/cp/convert/chronosQ/impl/redis/RedisSchedulerStore.java

```diff
@@ -100,44 +100,70 @@ public Long remove(String value, long time, int partitionNum) throws SchedulerEx
     }
 
     @Override
-    public List<String> get(long time, int partitionNum) throws SchedulerException {
+    public List<SchedulerEntry> get(long time, int partitionNum) throws SchedulerException {
         Jedis jedis = null;
+        List<SchedulerEntry> schedulerDataList;
         Set<String> resultSet;
         String key = "";
         try {
             jedis = _getInstance(partitionNum);
             key = getKey(time, partitionNum);
             resultSet = jedis.smembers(key);
             log.info("Get For " + key + "-" + resultSet);
+            schedulerDataList = getSchedulerPayloadValues(partitionNum, Optional.ofNullable(resultSet).orElse(Collections.emptySet()));
         } catch (Exception ex) {
             log.error("Exception occurred  for -" + "Key" + key + "Partition " + partitionNum + "-" + ex.getMessage());
             throw new SchedulerException(ex, ErrorCode.DATASTORE_READWRITE_ERROR);
         } finally {
             if ((jedis != null))
                 jedis.close();
         }
-        return new ArrayList<String>(resultSet);
+        return schedulerDataList;
     }
 
 
     @Override
-    public List<String> getNextN(long time, int partitionNum, int n) throws SchedulerException {
+    public List<SchedulerEntry> getNextN(long time, int partitionNum, int n) throws SchedulerException {
         Jedis jedis = null;
         List<String> resultSet;
+        List<SchedulerEntry> schedulerDataList;
         String key = "";
         try {
             jedis = _getInstance(partitionNum);
             key = getKey(time, partitionNum);
             resultSet = jedis.srandmember(key, n);
             log.info("Get For " + key + "-" + resultSet);
+            schedulerDataList = getSchedulerPayloadValues(partitionNum, Optional.ofNullable(resultSet).orElse(Collections.emptyList()));
         } catch (Exception ex) {
             log.error("Exception occurred  for -" + "Key" + key + "Partition " + partitionNum + "-" + ex.getMessage());
             throw new SchedulerException(ex, ErrorCode.DATASTORE_READWRITE_ERROR);
         } finally {
             if ((jedis != null))
                 jedis.close();
         }
-        return new ArrayList<String>(resultSet);
+        return schedulerDataList;
+    }
+
+    private List<SchedulerEntry> getSchedulerPayloadValues(int partitionNum, Collection<String> resultSet) throws SchedulerException {
+        Jedis jedis = null;
+        List<SchedulerEntry> schedulerDataList = new ArrayList<>();
+        try {
+            jedis = _getInstance(partitionNum);
+            List<String> schedulerValues = Optional.ofNullable(jedis.mget(resultSet.toArray(new String[resultSet.size()])))
+                    .orElse(Collections.emptyList());
+            Iterator<String> keyIterator = resultSet.iterator();
+            Iterator<String> valueIterator = schedulerValues.iterator();
+            while (keyIterator.hasNext() && valueIterator.hasNext()) {
```

Comment:
@utk4rsh @utk4rsh-fk There would be issue here when value iterator doesn't match keyIterator for existing keys, needs to be handled gracefully.

Reply:
my bad, mget would return nil elements